packages/core/src/list-transactions.ts

```diff
@@ -0,0 +1,144 @@
+import type { ArtifactResolver } from "./types/artifact";
+
+import { IgnitionError } from "./errors";
+import { FileDeploymentLoader } from "./internal/deployment-loader/file-deployment-loader";
+import { ERRORS } from "./internal/errors-list";
+import {
+  getExecutionOrder,
+  loadDeploymentState,
+} from "./internal/execution/deployment-state-helpers";
+import { ExecutionResultType } from "./internal/execution/types/execution-result";
+import {
+  ExecutionSateType,
+  type CallExecutionState,
+  type DeploymentExecutionState,
+  type ExecutionState,
+  type SendDataExecutionState,
+} from "./internal/execution/types/execution-state";
+import {
+  type Transaction,
+  TransactionReceiptStatus,
+} from "./internal/execution/types/jsonrpc";
+import { assertIgnitionInvariant } from "./internal/utils/assertions";
+import {
+  type ListTransactionsResult,
+  TransactionStatus,
+} from "./types/list-transactions";
+
+/**
+ * Returns the transactions associated with a deployment.
+ *
+ * @param deploymentDir - the directory of the deployment to get the transactions of
+ * @param artifactResolver - the artifact resolver to use when loading artifacts
+ * for a future
+ *
+ * @beta
+ */
+export async function listTransactions(
+  deploymentDir: string,
+  _artifactResolver: Omit<ArtifactResolver, "getBuildInfo">
+): Promise<ListTransactionsResult> {
+  const deploymentLoader = new FileDeploymentLoader(deploymentDir);
+
+  const deploymentState = await loadDeploymentState(deploymentLoader);
+
+  if (deploymentState === undefined) {
+    throw new IgnitionError(ERRORS.LIST_TRANSACTIONS.UNINITIALIZED_DEPLOYMENT, {
+      deploymentDir,
+    });
+  }
+
+  const executionOrder = await getExecutionOrder(deploymentLoader);
```

Comment:
I'm confused because I'm pretty sure the logic you wrote is doing the exact same thing as the logic I wrote? 

```
for (const journalMessage of journal) {
   if (!isSendTrasanction(journalMessage))  continue;
   const [future, networkInteraction, transaction] = somehowGetTheData(journalMessage);
   // same logic
}
```

```typescript
export async function getExecutionOrder(
  deploymentLoader: DeploymentLoader
): Promise<string[]> {
  const futureIds: string[] = [];

  // get each journal message in the order they were written, same as above
  for await (const message of deploymentLoader.readFromJournal()) {
    // ensures the message is relevant and not journal metadata. also makes sure each entry to the array is unique
    if ("futureId" in message && !futureIds.includes(message.futureId)) {
      futureIds.push(message.futureId);
    }
  }

  return futureIds;
}
```

```typescript
// iterating through the future IDs we just retrieved, in the same order they were written to the journal
 for (const futureId of executionOrder) {
    // this is the "somehowGetTheData" part of your snippet
    const exState = deploymentState.executionStates[futureId];

    // skip if this future doesn't send transactions, just like your snippet
    if (!doesSendTransactions(exState)) {
      continue;
    }
```

am I misunderstanding?

Reply:
update: i was in fact misunderstanding :)